plugins/modules/networkfirewall.py

```diff
@@ -0,0 +1,356 @@
+        configuration synchronization state.
+    type: bool
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - Maximum time, in seconds, to wait for the firewall to reach the
+        expected state.
+      - Defaults to 600 seconds.
+    type: int
+    required: false
+  subnets:
+    description:
+      - The ID of the subnets to which the firewall will be associated.
+      - Required when creating a new firewall.
+    type: list
+    elements: str
+    required: false
+  purge_subnets:
+    description:
+      - If I(purge_subnets=true), existing subnets will be removed from the
+        firewall as necessary to match exactly what is defined by I(subnets).
+    type: bool
+    required: false
+    default: true
+  policy:
+    description:
+      - The ARN of the Network Firewall policy to use for the firewall.
+      - Required when creating a new firewall.
+    type: str
+    required: false
+
+author: Mark Chappell (@tremble)
+extends_documentation_fragment:
+  - amazon.aws.aws
+  - amazon.aws.ec2
+'''
+
+EXAMPLES = '''
+# Create an AWS Network Firewall
+- community.aws.networkfirewall:
+    name: 'ExampleFirewall'
+    state: present
+    policy: 'ExamplePolicy'
+    subnets:
+    - 'subnet-123456789abcdef01'
+
+# Create an AWS Network Firewall with various options, don't wait for creation
+# to finish.
+- community.aws.networkfirewall:
+    name: 'ExampleFirewall'
+    state: present
+    delete_protection: True
+    description: "An example Description"
+    policy: 'ExamplePolicy'
+    policy_change_protection: True
+    subnets:
+    - 'subnet-123456789abcdef01'
+    - 'subnet-abcdef0123456789a'
+    subnet_change_protection: True
+    tags:
+      ExampleTag: Example Value
+      another_tag: another_example
+    wait: false
+
+
+# Delete an AWS Network Firewall
+- community.aws.networkfirewall:
+    state: absent
+    name: 'ExampleFirewall'
+'''
+
+RETURN = '''
+firewall:
+  description: The full details of the firewall
+  returned: success
+  type: dict
+  contains:
+    firewall:
+      description: The details of the firewall
+      type: dict
+      returned: success
+      contains:
+        delete_protection:
+          description: A flag indicating whether it is possible to delete the firewall.
+          type: str
+          returned: success
+          example: true
+        description:
+          description: A description of the firewall.
+          type: str
+          returned: success
+          example: "Description"
+        firewall_arn:
+          description: The ARN of the firewall.
+          type: str
+          returned: success
+          example: "arn:aws:network-firewall:us-east-1:123456789012:firewall/ExampleFirewall"
+        firewall_id:
+          description: A unique ID for the firewall.
+          type: str
+          returned: success
+          example: "12345678-abcd-1234-abcd-123456789abc"
+        firewall_name:
```

Comment:
or scrub `firewall_` prefix from the return keys.

The intention is, that you can keep `_info` parameter output as kind of a state file, and pass it into the none `_info` module.  
When you got hundrets of ressources, the roll out speeds up extremely, when you just diff the state file and the current `_info` return values and just apply the ones that changes.

Reply:
I don't like deviating too far from what boto3 itself returns.